### DISCUSSION TITLE
IntelliJ  插件兼容

### DISCUSSION THREAD

**Discussion by venciallee** [opensumi/core#2000] [General]
![image](https://user-images.githubusercontent.com/3282495/203330678-85849ff8-9075-4ec7-aedf-b1e23d072b60.png)
云栖大会提到 OpenSumi 的未来规划中, 包括兼容 IntelliJ 插件, 请问这部分能力能兼容所有插件还是部分兼容?

**Comment by bk1012** [comment]
这个我来解释下：

这个兼容可能跟你想的不大一样哈，现在主流的 IntelliJ 插件跟 OpenSumi 插件在运行时及开发语言上的差异基本上就是雷锋跟雷峰塔的关系了，如果用 OpenSumi 平台去兼容 IntelliJ 插件，只能说从技术角度来讲理论上可行，但成本会非常非常高，且多数情况下不可避免的需要去修改插件代码用来适配平台（两个平台在功能上是有差异的）

我们内部目前正在探索的方向是借助 JetBrains 平台内置的 JCEF 控件来做到两个平台插件架构的统一，这里这个兼容更多的指的是”插件代码一次编写，双平台交付“这么一个目标下的成果

**Reply by venciallee** [reply]
明白, 谢谢。站在 OpenSumi 角度确实不大好兼容 IntelliJ 插件, 不过要做一个 VSCode 体系的 IDE去兼容 IntelliJ 插件理论上是可行的, IntelliJ 插件运行环境问题是可以解决的, 我觉得可能成本比较高的是 IntelliJ + OpenSumi(VSCode 体系) 怎么对齐能力和相互怎么交互。
"插件代码一次编写，双平台交付"这个点, 在 IntelliJ + OpenSumi 能力上会对齐吗？

**Reply by bk1012** [reply]
是的，最大成本就是在能力对齐上，也确实无法做到百分百对齐

1. 这块的工作量在 JetBrains 这一侧比较大，我们优先会对齐内部使用到的一些能力，后续可能就要借助社区的力量来一起共建了。
2. 两个平台在功能上还是有差异的，如果一个能力在某个平台上压根就没有，这个也对齐不了
3. JCEF 这个技术受限制比较大，要求插件必须要有一个自定义界面。如果一个插件完全不依赖 UI 界面，那么想要跑在 OpenSumi/VSCode 下就得是另外一套方案了，目前这套方案还没有落地的计划

**Reply by venciallee** [reply]
👍, 期待 OpenSumi 的方案。另外 JetBrains Fleet 也在探讨怎么支持 IntelliJ 插件, Fleet 是直接把 IntelliJ IDEA 集成进去了, 运行环境和部分能力是没问题的, 不过还有其它能力、UI 交互(Fleet + IntelliJ) 不确定是怎么搞。

**Reply by bk1012** [reply]
那这个就看 Fleet 后面怎么做方案了，不过 Fleet 目前还不是主流的 IDE，我们一段时间内应该不会考虑 Fleet 这个平台~
